### PR TITLE
example on using ticks and grids added

### PR DESCRIPTION
**Pull request by rajasekharmekala**
This PR is continuation of #2372.The necessary changes mentioned in the comments on that PR  are modified here.

### CODE REVIEW

**Comment by nabobalis** [comment]
Does this mean I can close the other PR?

File: examples/finegrained_plot.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from astropy.coordinates import SkyCoord
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE
+
+
+###############################################################################
+# skyCoord module provides flexible celestial coordinate representation and a
```
Comment (nabobalis) [inline examples/finegrained_plot.py]: I think it should be `SkyCoord`

File: examples/finegrained_plot.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# draw_limb method overplots the boundary of the spacial objects.Date of the
```
Comment (nabobalis) [inline examples/finegrained_plot.py]: Space after the full stop

**Comment by nabobalis** [comment]
Can you create the example and embed the output here please?

**Comment by rajasekharmekala** [comment]
figure_1

**Comment by nabobalis** [comment]
Thanks! Looks like the title didn't render right.

**Comment by rajasekharmekala** [comment]
figure_1
Title corrected

**Comment by rajasekharmekala** [comment]
Have no idea.It works fine in python notebook.i think it's a problem with my libraries.It works fine in other systems
fig2
jupyter Notebook  image

**Comment by rajasekharmekala** [comment]
and i do not understand why travis-ci is failing

**Comment by nabobalis** [comment]
That looks better, well if it's just an issue with where you are running it, it should come out fine in the gallery. 

Which one is failing?

File: examples/finegrained_plot.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+# top and right edges.
+# The plots show the grid as white faint dashed lines by default and that can be removed.
+# Also the coordinates are float values for general usage but integers can be used for either axis.
+# For more information regarding the axis and grid settings, go to
+# http://docs.astropy.org/en/stable/visualization/wcsaxes/ticks_labels_grid.html .
+
+fig = plt.figure(figsize=(6, 6))
+ax = plt.subplot(projection=aiamap_sub)
+aiamap_sub.plot()
+aiamap_sub.draw_limb(color='black', linewidth=2, linestyle='dashed')
```
Comment (wafels) [inline examples/finegrained_plot.py]: Why is the limb black?  I would make it white.

File: examples/finegrained_plot.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+overlay = ax.get_coords_overlay('heliographic_stonyhurst')
```
Comment (wafels) [inline examples/finegrained_plot.py]: This is the important part, and needs an explanation.  I'd put a comment above this line.  "In order to gain fine-grained control of the Heliographic Stonyhurst grid, we pull it out of the Axes object and change its properties explicitly."

File: examples/finegrained_plot.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+lon.set_ticks_visible(False)
+lat.set_ticks_visible(False)
+lat.set_ticklabel_visible(False)
+lon.set_ticklabel_visible(False)
+
+lon.coord_wrap = 180
+lon.set_major_formatter('dd')
+
+overlay.grid(color='blue', linewidth=2, linestyle='dashed')
```
Comment (wafels) [inline examples/finegrained_plot.py]: Above this line add the comment "Plot the Heliographic Stonyhurst grid."

File: examples/finegrained_plot.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+ax.grid(False)
```
Comment (wafels) [inline examples/finegrained_plot.py]: Above this line, add the comment "Switch off the helioprojective Cartesian grid".

File: examples/finegrained_plot.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+tx, ty = ax.coords
```
Comment (wafels) [inline examples/finegrained_plot.py]: Above this line, add the comment "Change how the tick labels are formatted."

File: examples/finegrained_plot.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+aiamap = sunpy.map.Map(AIA_171_IMAGE)
+
+bottom_left = SkyCoord(-400*u.arcsec, -900*u.arcsec, frame=aiamap.coordinate_frame)
+top_right = SkyCoord(800*u.arcsec, 700*u.arcsec, frame=aiamap.coordinate_frame)
+aiamap_sub = aiamap.submap(bottom_left, top_right)
+
+title_obsdate = '{:%Y-%b-%d %H:%M:%S}'.format(aiamap_sub.date)
+
+###############################################################################
+# The figure displays coordinates on all four edges by default. The edges can be
```
Comment (wafels) [inline examples/finegrained_plot.py]: This is not true.  aiamap.peek() shows only a helioprojective Cartesian grid by default.  Please re-write to "The SunPy map peek method shows a helioprojective Cartesian grid by default.  This is sometimes not desired, and instead a heliographic Stonyhurst grid is desired.  Although maps offer a simple way to include a Stonyhurst grid, often times we want more control over the Stonyhusrt grid.  The example below illustrates some aspects of how a map plot can be altered as desired.  For more information regarding the axis and grid settings, go to http://docs.astropy.org/en/stable/visualization/wcsaxes/ticks_labels_grid.html .

File: examples/finegrained_plot.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE
+
+
+###############################################################################
+# SkyCoord module provides flexible celestial coordinate representation and a
+# draw_limb method overplots the boundary of the spacial objects. Date of the
```
Comment (nabobalis) [inline examples/finegrained_plot.py]: Maybe its due to a lack of my second coffee, but I am confused by the meaning of, `boundary of the spacial objects`

File: examples/finegrained_plot.py [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+bottom_left = SkyCoord(-400*u.arcsec, -900*u.arcsec, frame=aiamap.coordinate_frame)
+top_right = SkyCoord(800*u.arcsec, 700*u.arcsec, frame=aiamap.coordinate_frame)
+aiamap_sub = aiamap.submap(bottom_left, top_right)
+
+title_obsdate = '{:%Y-%b-%d %H:%M:%S}'.format(aiamap_sub.date)
+
+###############################################################################
+# The SunPy map peek method shows a helioprojective Cartesian grid by default.
+# This is sometimes not desired, and instead a heliographic Stonyhurst grid is
+# desired. Although maps offer a simple way to include a Stonyhurst grid, often
```
Comment (nabobalis) [inline examples/finegrained_plot.py]: `desired` used twice.

File: examples/finegrained_plot.py [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+# times we want more control over the Stonyhusrt grid. The example below illustrates
```
Comment (nabobalis) [inline examples/finegrained_plot.py]: Stonyhurst type.

File: examples/finegrained_plot.py [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+# regarding the axis and grid settings, go to
+# http://docs.astropy.org/en/stable/visualization/wcsaxes/ticks_labels_grid.html
+
+fig = plt.figure(figsize=(6, 6))
+ax = plt.subplot(projection=aiamap_sub)
+aiamap_sub.plot()
+aiamap_sub.draw_limb(color='white', linewidth=2, linestyle='dashed')
+
+# To have more sensitive control over the Heliographic Stonyhurst grid, the Axes
+# object is pulled out and change its properties explicitly.
```
Comment (nabobalis) [inline examples/finegrained_plot.py]: I have forgotten if these comments show up in the example without a 
`###############################################################################` above them.

**Comment by rajasekharmekala** [comment]
doc

regarding the single line comments, i tried building with sphinx and the page looks as above and shows the comments

**Comment by nabobalis** [comment]
If those comments are important they should be rendered in the html instead of in the python code block.

**Comment by wafels** [comment]
I don't think they need to be in the HTML.  They just need to be somewhere
to tell the user what particular lines are doing. Something like what
scikit-learn does

http://scikit-learn.org/stable/auto_examples/plot_johnson_lindenstrauss_bound.html#sphx-glr-auto-examples-plot-johnson-lindenstrauss-bound-py

**Comment by rajasekharmekala** [comment]
@wafels travis-ci fails for python3.6 for files which i did not edit and I do not know why..

**Comment by nabobalis** [comment]
That's due to a known problem, you can ignore that one.

**Comment by wafels** [comment]
I'd remove the word "sensitive" from "sensitive control".

File: examples/finegrained_plot.py [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+title_obsdate = '{:%Y-%b-%d %H:%M:%S}'.format(aiamap_sub.date)
+
+###############################################################################
+# The SunPy map peek method shows a helioprojective Cartesian grid by default.
+# This is sometimes not desired, and instead a heliographic Stonyhurst grid might
+# be required. Although maps offer a simple way to include a Stonyhurst grid, often
+# times we want more control over the Stonyhurst type. The example below illustrates
+# some aspects of how a map plot can be altered as desired. For more information
+# regarding the axis and grid settings, go to
+# http://docs.astropy.org/en/stable/visualization/wcsaxes/ticks_labels_grid.html
```
Comment (Cadair) [inline examples/finegrained_plot.py]: this should be formatted as an rst link i.e see 
```
`Ticks, tick labels and grid lines <http://docs.astropy.org/en/stable/visualization/wcsaxes/ticks_labels_grid.html>`_ in astropy
``

File: examples/finegrained_plot.py [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+aiamap = sunpy.map.Map(AIA_171_IMAGE)
+
+bottom_left = SkyCoord(-400*u.arcsec, -900*u.arcsec, frame=aiamap.coordinate_frame)
+top_right = SkyCoord(800*u.arcsec, 700*u.arcsec, frame=aiamap.coordinate_frame)
+aiamap_sub = aiamap.submap(bottom_left, top_right)
+
+title_obsdate = '{:%Y-%b-%d %H:%M:%S}'.format(aiamap_sub.date)
+
+###############################################################################
+# The SunPy map peek method shows a helioprojective Cartesian grid by default.
```
Comment (Cadair) [inline examples/finegrained_plot.py]: drop Cartesian

File: examples/finegrained_plot.py [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+
+lon.coord_wrap = 180
+lon.set_major_formatter('dd')
+
+# Plot the Heliographic Stonyhurst grid
+overlay.grid(color='blue', linewidth=2, linestyle='dashed')
+# Switch off the helioprojective Cartesian grid
+ax.grid(False)
+
+# Change how the helioprojective Cartesian grid tick labels are formatted
```
Comment (Cadair) [inline examples/finegrained_plot.py]: drop cartesian

File: examples/finegrained_plot.py [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+lat.set_ticks_visible(False)
+lat.set_ticklabel_visible(False)
+lon.set_ticklabel_visible(False)
+
+lon.coord_wrap = 180
+lon.set_major_formatter('dd')
+
+# Plot the Heliographic Stonyhurst grid
+overlay.grid(color='blue', linewidth=2, linestyle='dashed')
+# Switch off the helioprojective Cartesian grid
```
Comment (Cadair) [inline examples/finegrained_plot.py]: and here

**Comment by rajasekharmekala** [comment]
@nabobalis please suggest if any more changes are required

**Comment by nabobalis** [comment]
It looks fine to me.

**Comment by nabobalis** [comment]
Why did you revert that commit?

**Comment by rajasekharmekala** [comment]
@nabobalis  sorry.I was trying to fetch the new commits to my forked repository.How should i undo it?

**Comment by nabobalis** [comment]
Maybe try reverting the unrevert commit

**Comment by rajasekharmekala** [comment]
@nabobalis after you have approved the changes, i tried  fetch upstream from master.Then a commit occured.So i tried to revert it back.

**Comment by nabobalis** [comment]
Next time, you want to create work on branch from your fork instead of master. 

I think the only way now would be to rebase off upstream/master and delete the commits.

**Comment by rajasekharmekala** [comment]
@nabobalis is this acceptible?I think probably i should add a new PR..

**Comment by nabobalis** [comment]
Yeah, this isn't right. You seem to have added more commits and not deleted the revert commit.

**Pull request closed without merging**